charger/shelly.go

```diff
@@ -53,14 +55,19 @@ func NewShellyFromConfig(other map[string]interface{}) (api.Charger, error) {
 		return nil, errors.New("missing uri")
 	}
 
-	return NewShelly(cc.URI, cc.Channel, cc.StandbyPower)
+	return NewShelly(cc.URI, cc.User, cc.Password, cc.Channel, cc.StandbyPower)
 }
 
 // NewShelly creates Shelly charger
-func NewShelly(uri string, channel int, standbypower float64) (*Shelly, error) {
+func NewShelly(uri, user, password string, channel int, standbypower float64) (*Shelly, error) {
+	u, _ := url.Parse(uri)
```

Comment:
Fehlerbehandlung?

Reply:
Im Fehlerfall wäre u.Host == nil ... in dem Fall wird der uri string genommen wie er ist.